src/mcts/search.cc

```diff
@@ -907,6 +973,19 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend(
         ++possible_moves;
       }
       float Q = child.GetQ(fpu);
+
+      // Certainty Propagation. Avoid suboptimal childs.
+      if (params_.GetCertaintyPropagation()) {
+        // Prefers lower bounded childs over drawing children.
+        if (child.edge()->IsOnlyLBounded() && child.GetQ(0) <= 0.0f) Q = 0.01f;
```

Comment:
The size of these fake Q values seems to be a magic number.  Why 0.01 and not a smaller epsilon?

Reply:
Let me first get into the rational for this:
(1) If an edge is lower bounded we know that we at have least a (proven) draw if not more. Thus, if after search current average Q is below 0.0 this judgement (based on NN evals and search)  is guranteed to be wrong. 
(2) One mechanism to correct for these situations is that all Qs flowing through bounded edges get their scores corrected (see backpropagation), but this might take a while to adjust the score to be within bounds. Also it only converges to the "true" score (or bound), and if that happens from the wrong side, we misallocate visits. 
(3) In general, we would like to prefer lower bounded nodes (which are at least a draw, if not more) to drawing nodes.

The code above picks nodes that are lower-bounded with a virtual Q slightly higher than 0.00, if and only if the true score is out of proven bounds. As soon as the score is within bounds it can very well be 0.00. This speeds up the issues of (2) and encourages (3), while quickly restoring the natural equilibirum of the nodes Q and its lower bound.

As for the size of the virtual Q epsilon: In my testing it did not matter as long as its slightly larger than 0. But as these changes are subtle my testing was not rigerous enough to get something out of error bars for specific "epsilon" values. My explanation for this is that as soon as the real Q is back within bounds it has no further effect and the real Q is used, which can very well be 0.00. It seems that any "reasonable" value corrects the error in about the same number of visits (on average) and thus size is not that relevant. 
This "tweak" coincidenced with some other changes in PR487 so its hard to say what the differential elo impact is but the "final" version of PR487 (which includes this)  yielded better results than previous incarnations of CP. Moving from +10 to +20. But beware i am not claiming anything conclusive here - probably 0.001 or other sensible values yield similar results.